flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamIterationHead.java

```diff
@@ -71,10 +75,14 @@ public StreamIterationHead(Environment env) throws Exception {
 
     @Override
     protected void processInput(MailboxDefaultAction.Controller controller) throws Exception {
-        StreamRecord<OUT> nextRecord =
-                shouldWait
-                        ? dataChannel.poll(iterationWaitTime, TimeUnit.MILLISECONDS)
-                        : dataChannel.take();
+        StreamRecord<OUT> nextRecord;
```

Comment:
To add a UT, I plan to use `StreamTaskMailboxTestHarness` and trigger a checkpoint event in the mailbox. Is there any code example I can refer to? Thanks @tillrohrmann